### PR TITLE
Core/Scripts: Unify Chrono Literals

### PR DESCRIPTION
**Pull request by DanVS**
**Changes proposed:**
In continuation of #22415 unify Chrono Literals though the most of the TrinityCode code.

**Target branch(es):** 3.3.5

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp [hunk 1]
```diff
@@ -81,11 +81,11 @@ class boss_balinda : public CreatureScript
...
-            events.ScheduleEvent(EVENT_CONE_OF_COLD, 8 * IN_MILLISECONDS);
-            events.ScheduleEvent(EVENT_FIREBOLT, 1 * IN_MILLISECONDS);
-            events.ScheduleEvent(EVENT_FROSTBOLT, 4 * IN_MILLISECONDS);
-            events.ScheduleEvent(EVENT_SUMMON_WATER_ELEMENTAL, 3 * IN_MILLISECONDS);
-            events.ScheduleEvent(EVENT_CHECK_RESET, 5 * IN_MILLISECONDS);
+            events.ScheduleEvent(EVENT_CONE_OF_COLD, 8s);
+            events.ScheduleEvent(EVENT_FIREBOLT, 1s);
+            events.ScheduleEvent(EVENT_FROSTBOLT, 4s);
+            events.ScheduleEvent(EVENT_SUMMON_WATER_ELEMENTAL, 3s);
+            events.ScheduleEvent(EVENT_CHECK_RESET, 5s);
```
Comment (Faq) [inline src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp]: 5 * IN_MILLISECONDS rly?

Comment (DanVS) [reply src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp]: Excuse me?

**Comment by ccrs** [comment]
wait, how did you accomplish this? Dont tell me you did this with just text replace 😮

**Comment by DanVS** [comment]
Well, I did use them, but mostly semi-automatically/manually. A quick look showed that exact timings are saved.

I understand, this PR is hard to review. I might split it up into parts.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_overlord_wyrmthalak.cpp [hunk 2]
```diff
@@ -79,10 +79,10 @@ class boss_overlord_wyrmthalak : public CreatureScript
         void JustEngagedWith(Unit* /*who*/) override
         {
             _JustEngagedWith();
-            events.ScheduleEvent(EVENT_BLAST_WAVE, 20 * IN_MILLISECONDS);
+            events.ScheduleEvent(EVENT_BLAST_WAVE, 20s);
             events.ScheduleEvent(EVENT_SHOUT,       2 * IN_MILLISECONDS);
             events.ScheduleEvent(EVENT_CLEAVE,      6 * IN_MILLISECONDS);
```
Comment (Kittnz) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_overlord_wyrmthalak.cpp]: What about these?

Comment (Keader) [reply src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_overlord_wyrmthalak.cpp]: Nice spaces anyway xDD

File: src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp (hunk 1 above)
Comment (illfated) [reply src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp]: @Faq : you need to be more specific if you think something is wrong.

**Comment by HelloKitty** [comment]
Something like:
```
template<typename TChronoLiteralType1, typename TChronoLiteralType2>
uint32 urand(TChronoLiteralType1 chrono1, TChronoLiteralType2 chrono2)
{
    return urand(chrono1.count(), chorno2.count());
}
```

May be an approach to std::chorono literals with randomness. Though TC devs may prefer non-template overloads for each combination. It definitely does also assume you're working in reasonable time frames too since it truncates.

**Comment by Keader** [comment]
i dont get it, what is used for @HelloKitty ?

**Comment by HelloKitty** [comment]
@Keader Nothing, suggested it because based on the commits that left in all urand stuff it seemed like there wasn't a way to do random std::chrono literals. But there is already a way to register an event with them like that. So it was for nothing.

**Comment by Keader** [comment]
lol? it has.
like: 
```cpp
events.ScheduleEvent(EVENT_BLAST_WAVE, urand(20s,30s));
```
can be:
```cpp
events.ScheduleEvent(EVENT_BLAST_WAVE, 20s, 30s);
```

File: src/server/scripts/EasternKingdoms/AlteracValley/alterac_valley.cpp [hunk 3]
```diff
@@ -101,10 +101,10 @@ class npc_av_marshal_or_warmaster : public CreatureScript
                 events.Reset();
                 events.ScheduleEvent(EVENT_CHARGE_TARGET, urand(2 * IN_MILLISECONDS, 12 * IN_MILLISECONDS));
```
Comment (Keader) [inline src/server/scripts/EasternKingdoms/AlteracValley/alterac_valley.cpp]: what happens with those?
should be changed...
```events.ScheduleEvent(EVENT_CHARGE_TARGET, urand(2 * IN_MILLISECONDS, 12 * IN_MILLISECONDS));``` is the same of: ```events.ScheduleEvent(EVENT_CHARGE_TARGET, 2s, 12s);```

Comment (DanVS) [reply src/server/scripts/EasternKingdoms/AlteracValley/alterac_valley.cpp]: Should be fixed. Thanks!

File: src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp (hunk 1 above)
Comment (Faq) [reply src/server/scripts/EasternKingdoms/AlteracValley/boss_balinda.cpp]: Nevermind

**Comment by jackpoz** [comment]
is this ready to be merged ?

**Comment by Naios** [comment]
Great Job!

**Pull request merged**